core/loader/json-feed.ts

```diff
@@ -0,0 +1,112 @@
+import type { TextResponse } from '../download.js'
+import type { OriginPost } from '../post.js'
+import { createPostsPage } from '../posts-page.js'
+import type { Loader } from './index.js'
+import { findLinks, hasAnyFeed, toTime } from './utils.js'
+
+// https://www.jsonfeed.org/version/1.1/
+export type Author = {
+  avatar?: string
+  name?: string
+  url?: string
+}
+
+export type Item = {
+  /** deprecated from 1.1 version */
+  author?: Author
+  authors?: Author[]
+  banner_image?: string
+  content_html?: string
+  content_text?: string
+  date_modified?: string
+  date_published?: string
+  external_url?: string
+  id: string
+  image?: string
+  summary?: string
+  tags?: string[]
+  title?: string
+  url?: string
+}
+
+export type JsonFeed = {
+  /** deprecated from 1.1 version */
+  author?: Author
+  authors?: Author[]
+  description?: string
+  favicon?: string
+  feed_url?: string
+  home_page_url?: string
+  icon?: string
+  items?: Item[]
+  next_url?: string
+  title?: string
+  user_comment?: string
+  version: string
+}
+
+let existJsonFeedVersions = ['1', '1.1']
+
+function isValidJsonFeed(json: unknown): json is JsonFeed {
+  if (typeof json === 'object' && json !== null && 'version' in json) {
+    let ver = (json as JsonFeed).version.split('/').pop()
+    return existJsonFeedVersions.includes(ver!)
+  }
+  return false
+}
+
+function parsePosts(text: TextResponse): OriginPost[] {
+  let parsedJsonFeed = text.parseJson()
+  if (!isValidJsonFeed(parsedJsonFeed)) return []
+
+  return (
+    parsedJsonFeed.items?.map(item => ({
```

Comment:
The app will fall if `items` will be not an array ob objects.

We don’t have this problem in Atom/RSS loader because of XML API.

But since JSONs files in the global Internet could be in any form, we should be ready for broken JSONs.